crates/core/c8y_api/src/smartrest/inventory.rs

```diff
@@ -35,9 +47,37 @@ pub fn service_creation_message(
 ) -> Message {
     Message::new(
         &publish_topic_from_ancestors(ancestors),
+        // XXX: if any arguments contain commas, output will be wrong
         format!(
             "102,{},{},{},{}",
             service_id, service_type, service_name, service_status
         ),
     )
 }
+
+/// Create a SmartREST message for updating service status.
+///
+/// `service_status` can be any string, but `"up"`, `"down"`, and `"unknown"`
+/// have known meanings and are displayed in the UI in different ways.
+///
+/// `external_ids` differs from what is returned by `ancestors_external_ids` in
+/// that it also contains the external ID of the current entity (the one we want
+/// to set the status of).
+///
+/// https://cumulocity.com/guides/reference/smartrest-two/#104
+pub fn service_status_update_message(external_ids: &[String], service_status: &str) -> Message {
```

Comment:
To avoid any confusion, as described in the comments above, and to keep this API consistent with the other APIs in this module like `service_creation_message`, I'd just take the `service_id` and `ancestors` as separate parameters, the latter without the target `service_id` in it. But while passing it to `publish_topic_from_ancestors`, you'll have to append them.

Reply:
I'd like do do it properly later as part of a follow-up task.